doc/specs/stdlib_stats.md

```diff
@@ -166,6 +178,75 @@ program demo_mean
 end program demo_mean
 ```
 
+## `median` - median of array elements
+
+### Status
+
+Experimental
+
+### Description
+
+Returns the median of all the elements of `array`, or of the elements of `array` along dimension `dim` if provided, and if the corresponding element in `mask` is `true`.
+
+After that the elements are sorted in an increasing order, e.g. `array_sorted =
+sort(array)`, the median of the elements of `array` are defined as, if
+`n = size(array)` is an even number:
+
+```
+median(array) = array_sorted( floor( (n + 1) / 2.))
+```
+
+or as if `n` is an odd number:
+
+```
+median(array) = mean( array_sorted( floor( (n + 1) / 2.):floor( (n + 1) / 2.) + 1 ) )
+```
+
+The array is sorted using the subroutine `sort` provided by the `stdlib_sorting`
+module.
+
+### Syntax
+
+`result = [[stdlib_stats(module):median(interface)]](array [, mask])`
+
+`result = [[stdlib_stats(module):median(interface)]](array, dim [, mask])`
+
+### Class
+
+Generic subroutine
+
+### Arguments
+
+`array`: Shall be an array of type `integer` or  `real`.
+
+`dim`: Shall be a scalar of type `integer` with a value in the range from 1 to `n`, where `n` is the rank of `array`.
+
+`mask` (optional): Shall be of type `logical` and either a scalar or an array of the same shape as `array`.
+
+### Return value
+
+If `array` is of type `real`, the result is of the same type as `array`.
```

Comment:
Should this say  "... , the result is `real` with the same kind as the entries of `array`"?

And similarly for the integer case below?

Reply:
Agreed for the `real`.